src/Command/DependencyResolverCommand.php

```diff
@@ -158,4 +169,25 @@ private function parseExtensionBranches(string $extensionBranches): array
 
         return $extensionToBranchMap;
     }
+
+    /**
+     * Saves the generated composer.json into a file represented by its path.
+     *
+     * @param string $file
+     * @param string $composerJson
+     */
+    protected function resultToFile(string $file, string $composerJson): void
+    {
+        $dir = dirname($file);
+
+        if (!is_dir($dir)) {
+            if (false === mkdir($dir, 0755, true)) {
```

Comment:
Suggested solution to Race condition with mkdir should read:
```
        if (!mkdir($dir) && !is_dir($dir)) {
            throw new \RuntimeException(sprintf('Directory "%s" was not created', $dir));
        }
```

Reply:
Thanks Julien, indeed, it's better. 
I wonder what use case the lib may have could lead here 🤔 .